src/python/grpcio/commands.py

```diff
@@ -58,6 +63,117 @@ class CommandError(Exception):
   """Simple exception class for GRPC custom commands."""
 
 
+# TODO(atash): Remove this once PyPI has better Linux bdist support
+def _linux_target_egg_name(grpc_version):
+  """Get gRPC's custom made-up target names for Linux."""
+  assert sys.platform.startswith('linux')
+  version = '.'.join(map(str, sys.version_info[0:2]))
+  architecture = platform.machine()
+  flavor = 'ucs2' if sys.maxunicode == 65535 else 'ucs4'
+  return (
+      'grpcio-{grpc_version}-py{version}-linux-{architecture}-{flavor}.egg'
+        .format(grpc_version=grpc_version,
+                version=version,
+                architecture=architecture,
+                flavor=flavor))
+
+
+class Install(install.install):
+  """Custom Install command for gRPC Python.
+
+  This is for bdist shims and whatever else we might need a custom install
+  command for.
+  """
+
+  user_options = install.install.user_options + [
+      # TODO(atash): remove this once manylinux gets on PyPI
+      ('use-linux-bdist', None,
+       'Whether to retrieve a binary for Linux instead of building from '
+       'source.'),
+  ]
+
+  def initialize_options(self):
+    install.install.initialize_options(self)
+    self.use_linux_bdist = False
+
+  def finalize_options(self):
+    install.install.finalize_options(self)
+
+  def run(self):
+    if self.use_linux_bdist:
+      try:
+        egg_path = self._get_linux_bdist_egg()
+      except CommandError as error:
+        sys.stderr.write(
+            '\nWARNING: Failed to acquire grpcio prebuilt binary:\n'
+            '{}.\n\n'.format(error.message))
+        raise
+      try:
+        self._run_bdist_retrieval_install(egg_path)
+      except KeyboardInterrupt:
```

Comment:
I don't think this is necessary; `KeyboardInterrupt` and `Exception` are mutually exclusive. I think I may have given mistaken guidance in a code review a few weeks ago when what I really should have advised at the time was "avoid catching `Exception` unless you're really sure that you mean to catch things like programming defects as well as the exceptions the code under catch is documented to raise".